src/lib.rs

```diff
@@ -793,6 +802,16 @@ impl<T> Sender<T> {
     pub fn same_channel(&self, other: &Sender<T>) -> bool {
         Arc::ptr_eq(&self.shared, &other.shared)
     }
+
+    /// Discards excess capacity in the internal queue.
+    pub fn shrink_to_fit(&self) {
+        self.shared.shrink_to_fit();
+    }
+
+    /// Returns the number of elements the internal queue can hold without reallocating.
+    pub fn queue_capacity(&self) -> usize {
+        self.shared.queue_capacity()
+    }
```

Comment:
I don't think that `queue_capacity` should be externally exposed, the internal queue is very much an implementation detail.